app/src/main/java/com/example/unlibrary/unlibrary/UnlibraryViewModel.java

```diff
@@ -35,29 +53,113 @@ public UnlibraryViewModel() {
     }
 
     /**
-     * Gets a list of observable books. Typically from a repository.
+     * FailureMsgEvent getter for activity observers.
      *
-     * @return list of observable books
+     * @return Event of failure message to display
      */
-    public LiveData<List<Book>> getBooks() {
-        return mBooks;
+    public SingleLiveEvent<String> getFailureMsgEvent() {
+        return mFailureMsgEvent;
     }
 
     /**
-     * Cleans up resources, removes the snapshot listener from the repository.
+     * NavigationEvent getter for activity observers.
+     *
+     * @return Event of which fragment to navigate to
      */
-    @Override
-    protected void onCleared() {
-        mUnlibraryRepository.detachListeners();
+    public SingleLiveEvent<NavDirections> getNavigationEvent() {
+        return mNavigationEvent;
+    }
+
+
+    /**
+     * SuccessMsgEvent getter for activity observers.
+     *
+     * @return Event of which fragment to navigate to
+     */
+    public SingleLiveEvent<String> getSuccessMsgEvent() {
+        return mSuccessMsgEvent;
+    }
+
+    /**
+     * Getter for mCurrentBook live data object
+     *
+     * @return LiveData<Book> This returns the mBooks object
+     */
+    public LiveData<Book> getCurrentBook() {
+        return this.mCurrentBook;
```

Comment:
```suggestion
        return mCurrentBook;
```
nit: m prefix will always refer to this.memberVariable